codegen/facelift/templates/Struct.template.cpp

```diff
@@ -126,5 +131,77 @@ QString {{struct.name}}::toString() const
     return toStringWithFields(CLASS_ID, FIELD_NAMES);
 }
 
+#ifdef DBUS_IPC_ENABLED
```

Comment:
The structure definition headers are and should remain free from any IPC-related code.
Please create new headers for the DBus specific serialization code.

Reply:
I don't get the rationale for moving it out. Facelift is a DBus framework so its structure should now how to register themselves as dbus types and beside there are #ifdef guards for IPC_DISABLED builds.

Reply:
DBus is an implementation detail of Facelift's IPC layer, and Facelift can be built without DBus. No user code may depend on DBus being used or not, which means no definition related to DBus should be included when including a structure header. This is also a way to minimize the build time.

Reply:
There are ifdef guards for IPC_DISABLED so no QtDBus is included in that build variant.
If user build facelift with IPC_ENABLED obviously there will be IPC (QtDBus) related codes included, having those in header file or not has no effect on build time whatsoever. 

The following piece of code belongs logically to "struct" since it deals with struct's internal data structure and how to serialize it to QDBusArgument. So it can't and shouldn't moved to someplace else.

QDBusArgument &operator<<(QDBusArgument &argument, const {{struct.name}} &{{struct.name|lower}})

Reply:
Check the existing code to see how a struct is serialized without having to define the serialization function as "friend".

Reply:
I have already seen that IMHO wrong pattern which not only break the encapsulation of class by exposing all internal fields (you may as well define the facelift struct internal's public) via "asTuple" rather has a huge negative impact on performance. 
In my benchmarks construction of tuples where always one of the bottle-necks in IPC so I changed from std::tuple to https://github.com/taocpp/tuple/ and observed up to ~15-20% performance gain, probably higher gain still possible if one doesn't use tuples at all.
Anyhow I have switched back to the old "asTuple" way since I don't want to mix scopes of changes.